### ISSUE TITLE
Wrong window size in multimonitor display

### ISSUE THREAD

**Issue by salarcode** [0xd4d/dnSpy#333]
I have 2 monitors, 17" 96dpi and 15" 125dpi.
17" monitor is my main screen, so when I open any big modal window (Add Class, Edit Class, Edit Method(C#) and etc) the window is so big that falls out of my screen from all directions.
I think this happens because dnSpy uses my FHD monitor resolution to size itself.

Cause I couldn't screenshot it, here is a representation of it. The red line is the new window.
![monitor-position](https://cloud.githubusercontent.com/assets/1272095/21619749/4a7fbbf0-d206-11e6-8f50-b05e75c04a54.png)

**Comment by 0xd4d** [comment]
What is your screen resolution? dnSpy assumes everyone is using at least 1366x768 and the biggest dnSpy windows are 1300x730.

Workarounds: change the hard coded values and recompile or Alt+Space to bring up the system window menu.

The window sizes are hard coded in these files:

https://github.com/0xd4d/dnSpy/blob/master/Extensions/dnSpy.AsmEditor/Compiler/EditCodeDlg.xaml#L30
https://github.com/0xd4d/dnSpy/blob/master/Extensions/dnSpy.AsmEditor/MethodBody/MethodBodyDlg.xaml#L29

**Comment by salarcode** [comment]
My monitor resolutions are 1360 x 768 for 17" and 1920 x 1080 for 15".
I checked the window size with Spy++ and it showed that Edit Class window rectangle is 1625 x 913. This is 125% of the original hard coded size.

Clearly this issue is related dpi awareness per monitor, which maybe wpf chooses to scale itself according to the FHD monitor.

**Comment by 0xd4d** [comment]
Yes, that's correct. WPF pixels are logical and will increase or decrease in physical size depending on the DPI.
